### PR TITLE
Feature/best practices

### PR DESCRIPTION
**Pull request by domwoe**
Adds a Best Practices section to the Developer Docs with a first article on considerations for NFT developers.
I'm open to adding the content somewhere else.

- [x] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [x] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

### CODE REVIEW

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    * Use `Blob` instead of `[Nat8]` for storing large binary assets
+    * Consider using `Blob` instead of `[Nat8]` when sending or receiving Candid `vec nat8/blob` values. The choice is yours but `Blobs are 4x more compact and much less taxing on garbage collection (GC).
+    * Consider storing large `Blob`s in stable memory, to reduce pressure on the GC even further, especially when the manual memory management of that Blob is simple (e.g. they are only added, never deleted).
+    * Consider using the `compacting-gc` setting, especially in append-only scenarios, to allow access to larger heaps and reduce the cost of copying large, stationary objects.
+* Some advice for Rust developers:
+    * Be careful with extensive use of `Vec<u8>` and hence the `String` type if you need to (de-)serialize state for upgrades.
+    * Read [Roman’s blog post on effective Rust canisters](https://mmapped.blog/posts/01-effective-rust-canisters.html)
+
+A general article with [good practices for canister development by Joachim Breitner](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister).
```
Comment (sesi200) [inline docs/developer-docs/best-practices/considerations-for-nft-devs.md:60]: this is not a complete sentence

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    * Consider storing large `Blob`s in stable memory, to reduce pressure on the GC even further, especially when the manual memory management of that Blob is simple (e.g. they are only added, never deleted).
+    * Consider using the `compacting-gc` setting, especially in append-only scenarios, to allow access to larger heaps and reduce the cost of copying large, stationary objects.
+* Some advice for Rust developers:
+    * Be careful with extensive use of `Vec<u8>` and hence the `String` type if you need to (de-)serialize state for upgrades.
+    * Read [Roman’s blog post on effective Rust canisters](https://mmapped.blog/posts/01-effective-rust-canisters.html)
+
+A general article with [good practices for canister development by Joachim Breitner](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister). 
+
+To make sure you won’t get surprised, you can use the recently added [performance counter API](https://forum.dfinity.org/t/introducing-performance-counter-on-the-internet-computer/14027) to profile your canisters even before going live.
```
Comment (sesi200) [inline docs/developer-docs/best-practices/considerations-for-nft-devs.md:62]:
```suggestion
To make sure you won’t get surprised by ???, you can use the recently added [performance counter API](https://forum.dfinity.org/t/introducing-performance-counter-on-the-internet-computer/14027) to profile your canisters even before going live.
```

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+### Implement mechanisms to backup and restore state
+
+The IC itself does not yet support backup and restoration of the canister state, but it can be implemented in the canister itself. Regular backups are insurance against the worst-case scenario that a canister gets deallocated or there are issues with upgrading a canister.
```
Comment (sesi200) [inline docs/developer-docs/best-practices/considerations-for-nft-devs.md:67]: can you point to something that would explain how to do it?

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+### Consider using a dedicated service for storing the transaction history 
+
+There are dedicated services on the IC to keep an audit log of transactions such as [CAP](https://cap.ooo/), which can be used by an NFT collection as a service. This allows simple integration of the provenance history in explorers and wallets. 
+Furthermore, the state of ownership could be reconstructed in case the main NFT canister gets lost. However, some drawbacks have to be considered, e.g. NFT transfers incur additional costs due to the necessary inter-canister calls. 
+
+## Advanced Topics
+
+### Think about governance
+
+The value proposition of most NFTs is their permanence and immutability, e.g. by setting the [blackhole canister](https://github.com/ninegua/ic-blackhole) as the only controller.  As long as NFT canisters have their developers as controllers, users depend on the trustworthiness (and operational security) of the developers. Developers should therefore make the canisters immutable or manage the canisters with a DAO. A middle ground are mechanisms like [Launchtrail](https://devpost.com/software/launch-trail) that makes changes to a canister auditable.
```
Comment (sesi200) [inline docs/developer-docs/best-practices/considerations-for-nft-devs.md:79]:
```suggestion
The value proposition of most NFTs is their permanence and immutability, e.g. by setting the [blackhole canister](https://github.com/ninegua/ic-blackhole) as the only controller. As long as NFT canisters have their developers as controllers, users depend on the trustworthiness (and operational security) of the developers. Developers should therefore make the canisters immutable or manage the canisters with a DAO. A middle ground are mechanisms like [Launchtrail](https://devpost.com/software/launch-trail) that makes changes to a canister auditable.
```

Comment (sesi200) [inline docs/developer-docs/best-practices/considerations-for-nft-devs.md:79]:
```suggestion
The value proposition of most NFTs is their permanence and immutability, e.g. by setting the [blackhole canister](https://github.com/ninegua/ic-blackhole) as the only controller.  As long as NFT canisters have their developers as controllers, users depend on the trustworthiness (and operational security) of the developers. Developers should therefore make the canisters immutable or manage the canisters with a DAO. A middle ground are mechanisms like [Launchtrail](https://devpost.com/software/launch-trail) that make changes to a canister auditable.
```

File: docs/developer-docs/best-practices/index.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Best Practices
+
+In this section, we'll collect articles covering best practices and design patterns.
+
+For security-related best practices, follow [this link](../../references/security/index.md) to the reference section.
```
Comment (sesi200) [inline docs/developer-docs/best-practices/index.md:5]: missing link to NFT best practices

**Review by sesi200 — CHANGES_REQUESTED** [review]

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+# Considerations for NFT developers
+
+## What is an NFT?
+
+An NFT or Non-fungible Token is a record on a blockchain that is associated with a particular digital or physical asset. The unique digital representation on a blockchain allows the proving of ownership as well as their trading. 
+
+## NFTs on the Internet Computer
+
+The Internet Computer (IC) brings a lot of potential for NFTs. For digital assets like images, sound clips, or videos, the entire assets can live on-chain and can be included in on-chain games or metaverse experiences. Furthermore, we can imagine dynamic NFTs that change based on IC-internal and external data via [HTTPS outcalls](/https-outcalls).
```
[the remark this answers is not in the document]
Comment (domwoe) [reply docs/developer-docs/best-practices/considerations-for-nft-devs.md:9]: I tried this, but it's a `.tsx` file. How do I link to it?

[the remark this answers is not in the document]
Comment (sesi200) [reply docs/developer-docs/best-practices/considerations-for-nft-devs.md:9]: Of course I didn't think of that possibility... Then it's fine like that. I don't know a better way

File: docs/developer-docs/best-practices/considerations-for-nft-devs.md (hunk 3 above)
Comment (domwoe) [reply docs/developer-docs/best-practices/considerations-for-nft-devs.md:67]: Unfortunately, there's no resource on this yet.

Comment (sesi200) [reply docs/developer-docs/best-practices/considerations-for-nft-devs.md:67]: maybe this? https://forum.dfinity.org/t/backup-restore-function-for-a-canister/12849/3

Comment (domwoe) [reply docs/developer-docs/best-practices/considerations-for-nft-devs.md:67]: Good find!

**Review by sesi200 — APPROVED** [review]

**Pull request merged**